minecraft/resource/manifest.go

```diff
@@ -1,10 +1,55 @@
 package resource
 
-import "github.com/google/uuid"
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+)
 
 // Documentation on this may be found here:
 // https://learn.microsoft.com/en-us/minecraft/creator/reference/content/addonsreference/examples/addonmanifest
 
+// Version may be present in the manifest as [1,0,0] or "1.0.0".
+type Version [3]int
+
+func (v *Version) UnmarshalJSON(b []byte) error {
+	// Parse common array format [1,0,0]
+	var arr []int
+	if err := json.Unmarshal(b, &arr); err == nil && len(arr) > 0 {
+		for i := 0; i < 3 && i < len(arr); i++ {
+			v[i] = arr[i]
+		}
+		return nil
+	}
+
+	// Parse semver format "1.0.0"
+	var s string
+	if err := json.Unmarshal(b, &s); err == nil {
+		s = strings.TrimPrefix(strings.TrimSpace(s), "v")
+		if i := strings.IndexAny(s, "+-"); i >= 0 {
+			s = s[:i]
+		}
```

Comment:
We can probably skip this since their official parsing/validator wouldn't allow for this
https://github.com/Mojang/minecraft-creator-tools/blob/main/app/src/app/Package.ts#L202-L217